File: server/Models/Keep.cs
namespace keeper.Models;

public class Keep : DatabaseDefault
{
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public int Views { get; set; }
  public string CreatorId { get; set; }
  public Account Creator { get; set; }
  public int Kept { get; set; } // TODO figure out Kept (should be an int) amount of vaults its saved in. look at UML too.

}

public class KeepsInVault : Keep
{
  public int VaultKeepId { get; set; }
}

File: client/src/models/Keep.js
import { DatabaseDefault } from "./DatabaseDefault.js";

export class Keep extends DatabaseDefault {
constructor(d){
  super(d)
  this.name = d.name
  this.description = d.description
  this.img = d.img
  this.views = d.views
  this.creatorId = d.creatorId
  this.creator = d.creator
  this.kept = d.kept
}
}
export class VaultKeep extends Keep{
  constructor(d){
    super(d)
    this.vaultKeepId = d.vaultKeepId
  }
}